File: ruby/calendar.rb
require 'date'
require 'optparse'

date = DateTime.now
selected_month = Date.today.month

opt = OptionParser.new
opt.on('-m MONTH') do |v|
  month_number = v.to_i
  if month_number > 0 && month_number < 13
    selected_month = month_number
  else
    puts "#{v} is neither a month number (1..12) nor a name"
    exit(1)
  end
end

opt.parse(ARGV)

calender_year_date = date.strftime("#{selected_month}月 %Y年")
puts "     #{calender_year_date}"

a = Date.new(2023, selected_month, 1)
b = Date.new(2023, selected_month, -1)

numbers = []
(a..b).each do |date|
  numbers << date.mday
end

first_day = (a.wday + 6) % 7
week = %w[月 火 水 木 金 土 日]
puts week.join(' ')
print '   ' * first_day

numbers.each do |day|
  space = day.to_s.rjust(2)
  print "#{space} "
  puts if ((day + first_day) % 7).zero?
end

File: js/calendar.js
import { program } from "commander";

program
.argument('command')
.option('-m <month>', 'month', new Date().getMonth() + 1)
.parse(process.argv);

const options = program.opts();
const command = program.args[0];
const month = parseInt(options.m, 10);
if (month < 1 || month > 12) {
  console.error('不正な月です');
  process.exit(1);
}

const today = new Date();
const selected_month = month;
const calender_year_date = `${selected_month}月 ${today.getFullYear()}`;
const dayArr = ["日","月","火","水","木","金","土"];
const a = new Date(today.getFullYear(), selected_month - 1, 1);
const b = new Date(today.getFullYear(), selected_month, 0);
const firstDate = a.getDate();
const lastDate = b.getDate();
const firstDateDay = a.getDay();

let i = firstDate;
let result = '';
let j = 0;
while (j < firstDateDay) {
    result += '   ';
    j++;
}

while (i <= lastDate) {
    let formattedDay = i < 10 ? ' ' + i : i.toString();
    result += formattedDay + ' ';
    if ((i + firstDateDay) % 7 === 0) {
        result += '\n';
    }
    i++;
}

console.log(`      ${calender_year_date}\n${dayArr.join(' ')}\n${result}`);
